diff --git a/main.lua b/main.lua
index 1c99fcd4..9a24e43f 100644
--- a/main.lua
+++ b/main.lua
@@ -3,6 +3,7 @@ local rootPath = currentPath:gsub('[/\\]*[^/\\]-$', '')
 rootPath = (rootPath == '' and '.' or rootPath)
 loadfile(rootPath .. '/platform.lua')('script')
 local fs = require 'bee.filesystem'
+local math = require 'math'
 
 local function expanduser(path)
     if path:sub(1, 1) == '~' then
@@ -39,8 +40,9 @@ end
 loadArgs()
 
 ROOT     = fs.path(expanduser(rootPath))
-LOGPATH  = LOGPATH  and expanduser(LOGPATH)  or (ROOT:string() .. '/log')
-METAPATH = METAPATH and expanduser(METAPATH) or (ROOT:string() .. '/meta')
+TMP_ROOT = fs.path('/tmp/lua-language-server.' .. math.random(1000, 9000))
+LOGPATH  = LOGPATH  and expanduser(LOGPATH)  or (TMP_ROOT:string() .. '/log')
+METAPATH = METAPATH and expanduser(METAPATH) or (TMP_ROOT:string() .. '/meta')
 
 debug.setcstacklimit(200)
 collectgarbage('generational', 10, 50)
diff --git a/make.lua b/make.lua
index ce5c0012..fa8f0857 100644
--- a/make.lua
+++ b/make.lua
@@ -43,6 +43,4 @@ lm:build 'unittest' {
 
 lm:default {
     'install',
-    'test',
-    'unittest',
 }
diff --git a/platform.lua b/platform.lua
index 6c9d7c09..77e45c35 100644
--- a/platform.lua
+++ b/platform.lua
@@ -31,13 +31,7 @@ end
 
 local currentPath = debug.getinfo(1, 'S').source:sub(2)
 local fs = require 'bee.filesystem'
-local rootPath = fs.path(currentPath):remove_filename():string()
-if dll == '.dll' then
-    rootPath = rootPath:gsub('/', '\\')
-    package.path  = rootPath .. script .. '\\?.lua'
-          .. ';' .. rootPath .. script .. '\\?\\init.lua'
-else
-    rootPath = rootPath:gsub('\\', '/')
-    package.path  = rootPath .. script .. '/?.lua'
-          .. ';' .. rootPath .. script .. '/?/init.lua'
-end
+local rootPath = fs.path(currentPath):remove_filename()
+
+package.path = tostring(rootPath / script / '?.lua')
+     .. ';' .. tostring(rootPath / script / '?' / 'init.lua')
diff --git a/script/library.lua b/script/library.lua
index c7cecef6..afbc3a58 100644
--- a/script/library.lua
+++ b/script/library.lua
@@ -162,7 +162,7 @@ local function compileSingleMetaDoc(script, metaLang)
         end,
     }, { __index = _ENV })
 
-    util.saveFile((ROOT / 'log' / 'middleScript.lua'):string(), middleScript)
+    util.saveFile((TMP_ROOT / 'log' / 'middleScript.lua'):string(), middleScript)
 
     assert(load(middleScript, middleScript, 't', env))()
     if disable then
