diff --git a/src/completion/index.ts b/src/completion/index.ts
index c88a3661..c090f993 100644
--- a/src/completion/index.ts
+++ b/src/completion/index.ts
@@ -231,7 +231,8 @@ export class Completion implements Disposable {
           if (disableMenuShortcut && key == 'menu') {
             obj[key] = item[key].replace(/\[\w+\]$/, '')
           } else if (key == 'abbr' && item[key].length > labelMaxLength) {
-            obj[key] = item[key].slice(0, labelMaxLength)
+            obj[key] = item[key].slice(0, labelMaxLength - 3)
+            obj[key] += "..."
           } else {
             obj[key] = item[key]
           }
diff --git a/src/language-client/client.ts b/src/language-client/client.ts
index f36faf38..b3446d83 100644
--- a/src/language-client/client.ts
+++ b/src/language-client/client.ts
@@ -1895,7 +1895,7 @@ class CompletionItemFeature extends TextDocumentFeature<CompletionOptions, Compl
     const languageIds = cv.asLanguageIds(options.documentSelector!)
     const disposable = languages.registerCompletionItemProvider(
       this._client.id,
-      'LS',
+      'lsp',
       languageIds,
       provider,
       triggerCharacters,
diff --git a/src/languages.ts b/src/languages.ts
index c667dce4..e7219f83 100644
--- a/src/languages.ts
+++ b/src/languages.ts
@@ -736,7 +736,7 @@ class Languages {
       let detail = item.detail.replace(/\n\s*/g, ' ')
       if (byteLength(detail) < detailMaxLength) {
         if (detailField == 'menu') {
-          obj.menu = `${detail} ${obj.menu}`
+          obj.menu = `${obj.menu} ${detail}`
         } else if (detailField == 'abbr') {
           obj.abbr = `${obj.abbr} - ${detail}`
         }
