diff --git a/src/completion/index.ts b/src/completion/index.ts
index 4a90b21b..8a82dff8 100644
--- a/src/completion/index.ts
+++ b/src/completion/index.ts
@@ -211,7 +211,8 @@ export class Completion implements Disposable {
           if (disableMenuShortcut && key == 'menu') {
             obj[key] = item[key].replace(/\[.+\]$/, '')
           } else if (key == 'abbr' && item[key].length > labelMaxLength) {
-            obj[key] = item[key].slice(0, labelMaxLength)
+            obj[key] = item[key].slice(0, labelMaxLength - 3)
+            obj[key] += "..."
           } else {
             obj[key] = item[key]
           }
diff --git a/src/language-client/client.ts b/src/language-client/client.ts
index 8d303c99..90358ec1 100644
--- a/src/language-client/client.ts
+++ b/src/language-client/client.ts
@@ -1888,7 +1888,7 @@ class CompletionItemFeature extends TextDocumentFeature<CompletionOptions, Compl
     const languageIds = cv.asLanguageIds(options.documentSelector!)
     const disposable = languages.registerCompletionItemProvider(
       this._client.id + '-' + this.index,
-      'LS',
+      'lsp',
       languageIds,
       provider,
       triggerCharacters,
diff --git a/src/languages.ts b/src/languages.ts
index fc7aee36..663f1934 100644
--- a/src/languages.ts
+++ b/src/languages.ts
@@ -78,7 +78,11 @@ class Languages {
   private completionItemKindMap: Map<CompletionItemKind, string>
 
   constructor() {
-    this.loadCompleteConfig()
+    workspace.ready.then(() => {
+      this.loadCompleteConfig()
+    }, _e => {
+      // noop
+    })
     workspace.onDidChangeConfiguration(e => {
       if (e.affectsConfiguration('suggest')) {
         this.loadCompleteConfig()
@@ -695,7 +699,7 @@ class Languages {
       let detail = item.detail.replace(/\n\s*/g, ' ')
       if (byteLength(detail) < detailMaxLength) {
         if (detailField == 'menu') {
-          obj.menu = `${detail} ${obj.menu}`
+          obj.menu = `${obj.menu} ${detail}`
         } else if (detailField == 'abbr') {
           obj.abbr = `${obj.abbr} - ${detail}`
         }
