diff --git a/src/completion/index.ts b/src/completion/index.ts
index 364947da..33aceec8 100644
--- a/src/completion/index.ts
+++ b/src/completion/index.ts
@@ -231,7 +231,8 @@ export class Completion implements Disposable {
           if (disableMenuShortcut && key == 'menu') {
             obj[key] = item[key].replace(/\[\w+\]$/, '')
           } else if (key == 'abbr' && item[key].length > labelMaxLength) {
-            obj[key] = item[key].slice(0, labelMaxLength)
+            obj[key] = item[key].slice(0, labelMaxLength - 3)
+            obj[key] += "..."
           } else {
             obj[key] = item[key]
           }
diff --git a/src/language-client/client.ts b/src/language-client/client.ts
index 6bdda399..b5eb1557 100644
--- a/src/language-client/client.ts
+++ b/src/language-client/client.ts
@@ -1884,7 +1884,7 @@ class CompletionItemFeature extends TextDocumentFeature<CompletionOptions, Compl
     }
 
     const languageIds = cv.asLanguageIds(options.documentSelector!)
-    const disposable = languages.registerCompletionItemProvider(this._client.id, 'LS', languageIds, provider, triggerCharacters, allCommitCharacters)
+    const disposable = languages.registerCompletionItemProvider(this._client.id, 'lsp', languageIds, provider, triggerCharacters, allCommitCharacters)
     return [disposable, provider]
   }
 }
diff --git a/src/languages.ts b/src/languages.ts
index d3caa10e..144f9ca0 100644
--- a/src/languages.ts
+++ b/src/languages.ts
@@ -736,7 +736,7 @@ class Languages {
       let detail = item.detail.replace(/\n\s*/g, ' ')
       if (byteLength(detail) < detailMaxLength) {
         if (detailField == 'menu') {
-          obj.menu = `${detail} ${obj.menu}`
+          obj.menu = `${obj.menu} ${detail}`
         } else if (detailField == 'abbr') {
           obj.abbr = `${obj.abbr} - ${detail}`
         }
