diff --git a/src/completion/index.ts b/src/completion/index.ts
index f2fb6ca5..75637a9e 100644
--- a/src/completion/index.ts
+++ b/src/completion/index.ts
@@ -211,7 +211,8 @@ export class Completion implements Disposable {
           if (disableMenuShortcut && key == 'menu') {
             obj[key] = item[key].replace(/\[.+\]$/, '')
           } else if (key == 'abbr' && item[key].length > labelMaxLength) {
-            obj[key] = item[key].slice(0, labelMaxLength)
+            obj[key] = item[key].slice(0, labelMaxLength - 3)
+            obj[key] += "..."
           } else {
             obj[key] = item[key]
           }
diff --git a/src/language-client/client.ts b/src/language-client/client.ts
index 4d1073fe..130343a2 100644
--- a/src/language-client/client.ts
+++ b/src/language-client/client.ts
@@ -1886,7 +1886,7 @@ class CompletionItemFeature extends TextDocumentFeature<CompletionOptions, Compl
     const languageIds = cv.asLanguageIds(options.documentSelector!)
     const disposable = languages.registerCompletionItemProvider(
       this._client.id + '-' + this.index,
-      'LS',
+      'lsp',
       languageIds,
       provider,
       triggerCharacters,
diff --git a/src/languages.ts b/src/languages.ts
index 55014359..6e6a807f 100644
--- a/src/languages.ts
+++ b/src/languages.ts
@@ -118,7 +118,11 @@ class Languages {
         event.waitUntil(willSaveWaitUntil())
       }
     }, null, 'languageserver')
-    this.loadCompleteConfig()
+    workspace.ready.then(() => {
+      this.loadCompleteConfig()
+    }, _e => {
+      // noop
+    })
     workspace.onDidChangeConfiguration(this.loadCompleteConfig, this)
   }
 
@@ -769,7 +773,7 @@ class Languages {
       let detail = item.detail.replace(/\n\s*/g, ' ')
       if (byteLength(detail) < detailMaxLength) {
         if (detailField == 'menu') {
-          obj.menu = `${detail} ${obj.menu}`
+          obj.menu = `${obj.menu} ${detail}`
         } else if (detailField == 'abbr') {
           obj.abbr = `${obj.abbr} - ${detail}`
         }
