diff --git a/src/completion/index.ts b/src/completion/index.ts
index 205b08df..1accfe34 100644
--- a/src/completion/index.ts
+++ b/src/completion/index.ts
@@ -251,7 +251,8 @@ export class Completion implements Disposable {
           if (disableMenuShortcut && key == 'menu') {
             obj[key] = item[key].replace(/\[\w+\]$/, '')
           } else if (key == 'abbr' && item[key].length > labelMaxLength) {
-            obj[key] = item[key].slice(0, labelMaxLength)
+            obj[key] = item[key].slice(0, labelMaxLength - 3)
+            obj[key] += "..."
           } else {
             obj[key] = item[key]
           }
diff --git a/src/language-client/client.ts b/src/language-client/client.ts
index dde3ce67..278ca0fc 100644
--- a/src/language-client/client.ts
+++ b/src/language-client/client.ts
@@ -1895,7 +1895,7 @@ class CompletionItemFeature extends TextDocumentFeature<CompletionOptions, Compl
     const languageIds = cv.asLanguageIds(options.documentSelector!)
     const disposable = languages.registerCompletionItemProvider(
       this._client.id,
-      'LS',
+      'lsp',
       languageIds,
       provider,
       triggerCharacters,
diff --git a/src/languages.ts b/src/languages.ts
index ce6a3b62..bd5c695a 100644
--- a/src/languages.ts
+++ b/src/languages.ts
@@ -121,7 +121,11 @@ class Languages {
         event.waitUntil(willSaveWaitUntil())
       }
     }, null, 'languageserver')
-    this.loadCompleteConfig()
+    workspace.ready.then(() => {
+      this.loadCompleteConfig()
+    }, _e => {
+      // noop
+    })
     workspace.onDidChangeConfiguration(this.loadCompleteConfig, this)
   }
 
@@ -787,7 +791,7 @@ class Languages {
       let detail = item.detail.replace(/\n\s*/g, ' ')
       if (byteLength(detail) < detailMaxLength) {
         if (detailField == 'menu') {
-          obj.menu = `${detail} ${obj.menu}`
+          obj.menu = `${obj.menu} ${detail}`
         } else if (detailField == 'abbr') {
           obj.abbr = `${obj.abbr} - ${detail}`
         }
