diff --git a/autoload/fzf/vim.vim b/autoload/fzf/vim.vim
index d90c6f1..6fa9acf 100644
--- a/autoload/fzf/vim.vim
+++ b/autoload/fzf/vim.vim
@@ -749,8 +749,7 @@ function! fzf#vim#grep(grep_command, with_column, ...)
   \ 'source':  a:grep_command,
   \ 'column':  a:with_column,
   \ 'options': ['--ansi', '--prompt', capname.'> ',
-  \             '--multi', '--bind', 'alt-a:select-all,alt-d:deselect-all',
-  \             '--color', 'hl:4,hl+:12']
+  \             '--multi', '--bind', 'alt-a:select-all,alt-d:deselect-all']
   \}
   function! opts.sink(lines)
     return s:ag_handler(a:lines, self.column)
